python-threatexchange/CONTRIBUTING.md

```diff
@@ -12,9 +13,36 @@ black ./python-threatexchange/
 Additionally, your IDE may have support for automatically re-formatting your source files
 using black through your IDE settings.
 
-### Running Tests
-python-threatexchange is a bit short on tests, but you could help fix that.
-To run the tests `make test`
+When in doubt, follow the style of the file you are editing, and if starting new files, be consistent within.
+
+## python typing and `mypy`
+We aiming for full strict coverage via PEP484 tying annotations. However, this is a work in progress. We use [mypy](https://mypy.readthedocs.io/en/stable/index.html) to verifying typing.
+
+### `# type: ignore`
+We're not all typing experts yet, and we've confused ourselves with Generics, Aliases and object heirarchies, and occasionally `# type: ignored`. If you end up in a sticky situation and decide to throw in the towel on typing and use `# type: ignore`, use `mypy --show-error-codes` to only ignore the specific error code, and expect discussion on the PR on the level of effort needed to make the code compliant.
+
+### New Files or Directories
+When creating new files or directories, please first try and make them compatability with `--strict` to do this, add in the mypy.ini file
+```
+[path.to.module]
```

Comment:
Can we do the reverse? Add all current files with strict=False (one time) and expect all new files to be strict=True?

Reply:
Basically all files fail strict right now :/ I tried adding the strict flags one at a time in #1086, but found we're hundreds of errors away, and there's not a super clean way to tag all the files. I figured it'd be easy to do one directory at a time.